react/src/components/atoms/forms/Form/Form.stories.js

```diff
@@ -97,35 +79,78 @@ storiesOf('atoms/forms', module)
                   }
+                ],
+                [
+                  'test2',
+                  {
+                    ...inputTextOptionsWithKnobs,
+                    key: 'Form.InputNumber.test2',
+                    defaultValue: 100,
+                    labelText: 'Input 2 (Linked to Input 0, Input 1, and Slider)',
+                    id: 'test2',
+                    unit: '%'
+                  }
                 ]
               ];
               const inputs = [];
               ids.forEach((numberProps) => {
                 inputs.push(<InputNumber {...numberProps[1]} />);
               });
-              inputSliderOptionsWithKnobs.onUpdate = (newVal, id) => {
-                if (formContext.hasId(id)) {
-                  formContext.setValue({ id: 'test0', value: Number(formContext.getValue(id) * 100).toFixed() });
-                  formContext.setValue({ id: 'test1', value: Number((1 - formContext.getValue(id)) * 100).toFixed() });
-                  if (formContext.hasId('currency-input')) {
-                    if (newVal > 0.6) {
-                      // Sets currency to 999 when slider is 60%.
-                      formContext.setValue({ id: 'currency-input', value: '$999.00' });
-                    } else {
-                      formContext.setValue({ id: 'currency-input', value: '$0.00' });
-                    }
-                  }
+              // Custom conditions for when InputSync updates.
+              // InputSync will normally update whenever the form ids in formIds update.
+              // InputSync can only watch form ids that actually exist,
+              // so it must come after the input it is watching has been rendered.
+              const inputSyncProps = {
+                syncCondition: (updatedFormId, formValue) => {
+                  const currencyValue = Number(numbro.unformat(formContext.getValue('currency-input')));
+                  return(currencyValue > 5);
+                },
+                formIds: ['currency-input', 'test0']
+              };
+              // Using overrideLink breaks the Input's link to other Inputs.
+              // The returned value will become the new value for this Input instead.
+              inputSliderOptionsWithKnobs.overrideLink = (newVal) => {
+                return(Number(numbro(newVal / 100).format({ mantissa: 2 })));
+              };
+              // We still want the slider to change the other Inputs, so add an update function.
+              // Update the inputs back to percentages when slider changes value.
+              inputSliderOptionsWithKnobs.updateFunc = (val) => {
+                const newVal = Number(numbro(val * 100).format({ mantissa: 0 }));
+                // We need to ensure that the new percentage value isn't already set on the Inputs we are changing.
+                // If this check is not done, an infinite loop will occur.
+                if (!deepEqual(newVal, formContext.getValue('test0'))) {
```

Comment:
this means on the project side, we need to add the deep-equal package and do this check?